From a9755f3442316106dc7dc6eb85bb96dd4c38bf05 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Thu, 6 May 2010 13:15:00 +0800
Subject: [PATCH 153/435] ENGR00122871 Change vpu unit test to sync with
 v1.3.10 f/w on mx51 and mx53

Change vpu unit test to sync with v1.3.10 f/w on mx51 and mx53

Signed-off-by: Sammy He < r62914@freescale.com>
---
 test/mxc_vpu_test/dec.c | 6 ++++++
 test/mxc_vpu_test/enc.c | 9 +++++++++
 2 files changed, 15 insertions(+)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 659964f..490e08e 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -806,6 +806,12 @@ decoder_start(struct decode *dec)
 			vpu_DecGiveCommand(handle, SET_DBK_OFFSET, &dbkOffset);
 		}
 
+		/* RTC setting */
+		decparam.underrunEnable = 0;
+		decparam.accumulativeMode = 0;
+		decparam.pulseWidth = 0;
+		decparam.refMbClk = 0;
+
 		gettimeofday(&tdec_begin, NULL);
 		ret = vpu_DecStartOneFrame(handle, &decparam);
 		if (ret != RETCODE_SUCCESS) {
diff --git a/test/mxc_vpu_test/enc.c b/test/mxc_vpu_test/enc.c
index faec402..0c07321 100644
--- a/test/mxc_vpu_test/enc.c
+++ b/test/mxc_vpu_test/enc.c
@@ -456,6 +456,12 @@ encoder_start(struct encode *enc)
 				break;
 		}
 
+		/* RTC setting */
+		enc_param.underrunEnable = 0;
+		enc_param.accumulativeMode = 0;
+		enc_param.pulseWidth = 0;
+		enc_param.refMbClk = 0;
+
 		gettimeofday(&tenc_begin, NULL);
 		ret = vpu_EncStartOneFrame(handle, &enc_param);
 		if (ret != RETCODE_SUCCESS) {
@@ -725,10 +731,13 @@ encoder_open(struct encode *enc)
 	encop.rcIntraQp = -1;
 	encop.userQpMax = 0;
 	encop.userQpMin = 0;
+	encop.userQpMinEnable = 0;
+	encop.userQpMaxEnable = 0;
 
 	encop.userGamma = (Uint32)(0.75*32768);         /*  (0*32768 <= gamma <= 1*32768) */
 	encop.RcIntervalMode= 1;        /* 0:normal, 1:frame_level, 2:slice_level, 3: user defined Mb_level */
 	encop.MbInterval = 0;
+	encop.avcIntra16x16OnlyModeEnable = 0;
 
 	encop.ringBufferEnable = 0;
 	encop.dynamicAllocEnable = 0;
-- 
1.8.0

